### PR TITLE
Implemented columnValuesNotNullConstraint

### PR DESCRIPTION
**Pull request by MilenaTrajanoska**
## Description

Implemented columnValuesNotNullConstraint

### General Checklist

* [ ] Tests added for this feature/bug
      if it was a bug, test must cover it.
* [ ] Conform by the style guides, by using formatter
* [ ] Documentation updated
* [ ] (optional) Please add a label to your PR

### CODE REVIEW

**Review by Taxuspt — APPROVED** [review]

**Review by jamie256 — CHANGES_REQUESTED** [review]
Approach looks good, can you update to target mainline?

**Review by jamie256 — CHANGES_REQUESTED** [review]
Few more conflicts to resolve.

File: tests/unit/core/statistics/test_constraints.py [hunk 1]
```diff
@@ -1018,3 +1019,81 @@ def test_quantile_between_wrong_datatype():
...
+    profile = session.log_dataframe(df_lending_club, "test.data", constraints=dc)
+    session.close()
+    report = profile.apply_summary_constraints()
+
+    TEST_LOGGER.info(f"Apply columnValuesNotNullConstraint report:\n{report}")
+
+    assert report[0][1][0][0] == f"summary null_count {Op.Name(Op.EQ)} 0/None"
+    assert report[0][1][0][1] == 1
+    assert report[0][1][0][2] == 0
+
```
Comment (richard-rogers) [inline tests/unit/core/statistics/test_constraints.py:1319]: Split this into a separate test?

File: tests/unit/core/statistics/test_constraints.py [hunk 2]
```diff
@@ -1018,3 +1019,81 @@ def test_quantile_between_wrong_datatype():
         quantileBetweenConstraint(quantile_value=0.3, lower_value=1.24, upper_value=[6.63], verbose=True)
     with pytest.raises(ValueError):
         quantileBetweenConstraint(quantile_value=0.3, lower_value=2.3, upper_value=1.5, verbose=True)
+
+
+def test_column_values_not_null_constraint_apply(local_config_path, df_lending_club):
+    nnc1 = columnValuesNotNullConstraint()
+    nnc2 = columnValuesNotNullConstraint()
+
+    dc = DatasetConstraints(None, summary_constraints={"annual_inc": [nnc1]})
```
Comment (richard-rogers) [inline tests/unit/core/statistics/test_constraints.py]: This stanza is repeated a lot. Perhaps make it a fixture or function.

**Review by richard-rogers — CHANGES_REQUESTED** [review]

**Review by jamie256 — APPROVED** [review]

**Review by richard-rogers — APPROVED** [review]

**Pull request merged**